src/MaxText/input_pipeline/instruction_data_processing.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Preprocessing for instruction dataset."""
+
+import datasets
+
+REASONING_START = "<reasoning>"
+REASONING_END = "</reasoning>"
+ANSWER_START = "<answer>"
+ANSWER_END = "</answer>"
+
+QA_PROMPT_TEMPLATE = (
```

Comment:
Since this is specific to dataset/task. Should we make it easily configurable by the user? Maybe in a yaml file?